### ISSUE TITLE
Recognize new import takes too long in large repos

### ISSUE THREAD

**Issue by yusuphgammut** [typescript-language-server/typescript-language-server#675]
Hi folks.

__TLDR:__ The server is taking more than 1 minute to recognize new exports on modules, and only after the file that is making the export is saved. This happens in both Neovim and [Helix](https://helix-editor.com/), so it seems that isn't an issue with a specific configuration.

Neovim config (just in case):
``` lua
local nvim_lsp = require('lspconfig')

nvim_lsp.tsserver.setup{
  on_attach = on_attach,
  capabilities = require('cmp_nvim_lsp').default_capabilities(),
  flags = {
    debounce_text_changes = 150,
  },
  maxTsServerMemory = 4096,
  init_options = {
    preferences = {
      disableSuggestions = true,
    }
  }
}
```

__Neovim version:__ 0.8.1
__Node version:__ 16.10.0
__typescript-language-server version:__ 3.0.3
__typescript version:__ 4.9.4

__More context:__

In the company I work for there's a Typescript repository that's growing very fast. They migrated recently to [Nx](https://nx.dev/) build system, and the repository is quite large at the moment. A few weeks ago I updated some of the plugins and language servers in my Neovim configuration and I started noticing that in this repo there was quite a huge delay in some of the typescript-language-server diagnostics, especially when adding a new export in a module, and then trying to import it from other module.

If I don't save the file where the export is added, and try to import the new item in another file, then the server never recognizes that the module is exporting that new item, and the `Module 'example/file' has no exported member 'newItem'` error never hides. That wasn't happening before the update.

Then, if I save the file, the error `Module 'example/file' has no exported member 'newItem'` in the file that is importing is only hidden after a minute and half. This means that the server takes more than a minute to recognize the new export, which is clearly too long.

I tried this in both Neovim and Helix, with the same result (even after adding the `maxTsServerMemory` config). I then downloaded VSCode and did the same test with the out of the box configuration. It recognizes the new import almost instantly and without saving the file that make the export.

Am I missing a configuration in the server to improve this in large repos?

**Comment by rchl** [comment]
The speed can be greatly affected by which files are opened and how many. So unless you are testing with same files opened in all the editors then there can be big differences.

But I don't think I'll be able to help much with just those information and without a public repo that reproduces.

> That wasn't happening before the update.

You should be able to downgrade easily. And if actually can't reproduce in older version then it would help to specify when exactly this has regressed.

**Comment by yusuphgammut** [comment]
I understand. I'll try to downgrade to check with older versions and come back here.

**Comment by yusuphgammut** [comment]
Hi @rchl . I tried to reproduce the issue with previous versions of this server (v1.2.0) but it's still happening in my company's repo (using Neovim). I also downloaded the VSCode's repo and triplicated the codebase src folder to have around 10 thousand files, but the problem doesn't happen there (still using Neovim). So I guess it might be an issue with the structure of Nx projects or something like that. If I open my company's repo with VSCode the issue doesn't happen.

I did tried to replicate the problem and have a look at the logs with `logVerbosity = 'terse'` and `trace = 'messages'` in the `tsserver` options. This is the result just in case it might give you some light on the issue. What I did was:

1. Open files `/src/constants.ts` and `/src/component.tsx`.
2. In the constants file, write `export const mybla = "bla";`.
3. In the component file, write `import { mybla } from "src/constants";`. I get the message `Module 'src/constants' has no exported member 'mybla'`.
4. Wait a minute. Nothing happens, so then I save the constants file.
5. Wait between 3 to 6 minutes and then the diagnostics message disappears in the component file.

``` txt
----------------------------------------------------------------
TIMELINE
----------------------------------------------------------------

constants.ts             :    component.tsx
-------------------------:-----------------
 |                       :     |
[open file]              :    [open file]
 |                       :     |
Info 0 to Info 23        :    Info 0 to Info 23
 |                       :     |
[write export const]     :     |
 |                       :     |
Info 24 to Info 112      :     |
 |                       :     |
 |                       :    [write import const]
 |                       :     |
 |                       :    Info 24 to Info 47
 |                       :     |
[save file]              :     |
 |                       :     |
Info 113 to Info 143     :    Info 48 to Info 78
 |                       :     |
 |                       :    [... 5 minutes later] <-- Problem here!
 |                       :     |
 |                       :    Info 79 to Info 86
```

Here are the tsserver logs with some comments and edited parts.

For the constants file:
``` txt
// Open file.

Info 0    [10:27:41.739] Starting TS Server
Info 1    [10:27:41.741] Version: 4.8.4
Info 2    [10:27:41.741] Arguments: /Users/myUser/.nvm/versions/node/v16.10.0/bin/node /Users/myUser/myHugeProj/node_modules/typescript/lib/tsserver.js --useInferredProjectPerProjectRoot --cancellationPipeName /private/var/folders/lb/c400zl310zb_qp6vjb2dtgj40000gn/T/5b93f2c56092ac72308863777f80fec4/tscancellation* --logVerbosity terse --logFile /Users/myUser/.mybeautifullogs/tsserver-log-f45x7V/tsserver.log --locale en --validateDefaultNpmLocation
Info 3    [10:27:41.741] Platform: darwin NodeVersion: 16 CaseSensitive: false
Info 4    [10:27:41.741] ServerMode: undefined syntaxOnly: false hasUnknownServerMode: undefined
Info 5    [10:27:41.753] Host information neovim
Info 6    [10:27:41.754] Format host information updated
Info 7    [10:27:41.756] Search path: /Users/myUser/myHugeProj/src/constants
Info 8    [10:27:41.757] For info: /Users/myUser/myHugeProj/src/constants.ts :: Config file name: /Users/myUser/myHugeProj/tsconfig.json
Info 9    [10:27:41.757] Creating configuration project /Users/myUser/myHugeProj/tsconfig.json
Info 10   [10:27:41.770] Config: /Users/myUser/myHugeProj/tsconfig.json : {
 "rootNames": [],
 "options": {
  "rootDir": "/Users/myUser/myHugeProj",
  "sourceMap": true,
  "declaration": false,
  "moduleResolution": 2,
  "emitDecoratorMetadata": true,
  "experimentalDecorators": true,
  "importHelpers": true,
  "target": 2,
  "module": 99,
  "lib": [
   "lib.dom.d.ts",
   "lib.dom.iterable.d.ts",
   "lib.esnext.d.ts"
  ],
  "skipLibCheck": true,
  "skipDefaultLibCheck": true,
  "baseUrl": "/Users/myUser/myHugeProj",
  "paths": {
   "@myCompany/org-myCompany-platform-fn": [
    "libs/org/myCompany/platform/fn/src/index.ts"
   ],
   "@myCompany/myCompanyRouter": [
    "libs/org/myCompany/platform/router/src/index.ts"
   ],
   "*": [
    "src/app/*"
   ]
  },
  "pathsBasePath": "/Users/myUser/myHugeProj",
  "jsx": 4,
  "allowJs": true,
  "esModuleInterop": true,
  "allowSyntheticDefaultImports": true,
  "forceConsistentCasingInFileNames": true,
  "strict": true,
  "noImplicitOverride": false,
  "noPropertyAccessFromIndexSignature": false,
  "noImplicitReturns": false,
  "noFallthroughCasesInSwitch": true,
  "resolveJsonModule": true,
  "types": [
   "node",
   "jest",
   "facebook-js-sdk",
   "segment-analytics"
  ],
  "configFilePath": "/Users/myUser/myHugeProj/tsconfig.json"
 },
 "projectReferences": [
  {
   "path": "/Users/myUser/myHugeProj/tsconfig.app.json",
   "originalPath": "./tsconfig.app.json"
  },
  {
   "path": "/Users/myUser/myHugeProj/tsconfig.spec.json",
   "originalPath": "./tsconfig.spec.json"
  },
  {
   "path": "/Users/myUser/myHugeProj/.storybook/tsconfig.json",
   "originalPath": "./.storybook/tsconfig.json"
  }
 ]
}
Info 11   [10:27:41.772] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.json
Info 12   [10:27:44.808] Config: /Users/myUser/myHugeProj/tsconfig.app.json : {
  // Same config as above.
}
Info 13   [10:27:47.342] Config: /Users/myUser/myHugeProj/tsconfig.spec.json : {
  // Same config as above.
}
Info 14   [10:27:49.051] Config: /Users/myUser/myHugeProj/.storybook/tsconfig.json : {
  // Probably not important here.
}
Info 15   [10:27:49.060] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.json Version: 1 structureChanged: true structureIsReused:: Not Elapsed: 7288.094790995121ms
Info 16   [10:27:49.060] Different program with same set of files
Info 17   [10:27:49.080] Creating configuration project /Users/myUser/myHugeProj/tsconfig.app.json
Info 18   [10:27:49.805] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 19   [10:28:10.287] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 1 structureChanged: true structureIsReused:: Not Elapsed: 20482.08679100871ms
Info 20   [10:28:10.287] Project '/Users/myUser/myHugeProj/tsconfig.app.json' (Configured)
Info 21   [10:28:10.287] 	Files (8548)

Info 22   [10:28:10.287] -----------------------------------------------
Info 23   [10:28:10.374] AutoImportProviderProject: attempted to add more than 10 dependencies. Aborting.

// Write: export const mybla = "bla";

Info 24   [10:29:18.126] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 25   [10:29:18.510] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 2 structureChanged: false structureIsReused:: Completely Elapsed: 384.332415997982ms
Info 26   [10:29:18.510] Different program with same set of files
Info 27   [10:29:18.511] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 28   [10:29:18.783] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 3 structureChanged: false structureIsReused:: Completely Elapsed: 271.92174500226974ms
Info 29   [10:29:18.783] Different program with same set of files
Info 30   [10:29:18.802] getCompletionData: Get current token: 0.080144003033638
·
·
·
Info 107  [10:29:26.131] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 108  [10:29:26.402] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 16 structureChanged: false structureIsReused:: Completely Elapsed: 270.9055729955435ms
Info 109  [10:29:26.402] Different program with same set of files
Info 110  [10:29:27.211] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 111  [10:29:27.505] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 17 structureChanged: false structureIsReused:: Completely Elapsed: 294.2276470065117ms
Info 112  [10:29:27.505] Different program with same set of files

// Save file.

Info 113  [10:37:59.356] DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/4913 :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
Info 114  [10:37:59.362] Elapsed:: 5.097203001379967ms DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/4913 :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
Info 115  [10:37:59.362] DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/constants.ts :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
Info 116  [10:37:59.362] Elapsed:: 0.585095003247261ms DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/constants.ts :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
Info 117  [10:37:59.362] DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/constants.ts :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
Info 118  [10:37:59.363] Elapsed:: 0.41933900117874146ms DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/constants.ts :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
·
·
·
Info 135  [10:37:59.366] DirectoryWatcher:: Triggered with /Users/myUser/myHugeProj/src/constants.ts :: WatchInfo: /Users/myUser/myHugeProj/src 1 undefined Project: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Failed Lookup Locations
Info 136  [10:37:59.366] Elapsed:: 0.09715300798416138ms DirectoryWatcher:: Triggered with /Users/myUser/myHugeProj/src/constants.ts :: WatchInfo: /Users/myUser/myHugeProj/src 1 undefined Project: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Failed Lookup Locations
Info 137  [10:38:00.152] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 138  [10:38:00.179] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 18 structureChanged: false structureIsReused:: Completely Elapsed: 26.2674959897995ms
Info 139  [10:38:00.179] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.json
Info 140  [10:38:01.206] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.json Version: 2 structureChanged: false structureIsReused:: Not Elapsed: 1027.457734003663ms
Info 141  [10:38:01.865] Before ensureProjectForOpenFiles:
Info 142  [10:38:01.865] After ensureProjectForOpenFiles:
Info 143  [10:38:01.865] got projects updated in background, updating diagnostics for /Users/myUser/myHugeProj/src/constants.ts
```

For the component file:
``` txt
// Open file.

Info 0    [10:27:53.673] Starting TS Server
Info 1    [10:27:53.675] Version: 4.8.4
Info 2    [10:27:53.675] Arguments: /Users/myUser/.nvm/versions/node/v16.10.0/bin/node /Users/myUser/myHugeProj/node_modules/typescript/lib/tsserver.js --useInferredProjectPerProjectRoot --cancellationPipeName /private/var/folders/lb/c400zl310zb_qp6vjb2dtgj40000gn/T/0db405c0a573b26d700711b42bbf40f0/tscancellation* --logVerbosity terse --logFile /Users/myUser/.mybeautifullogs/tsserver-log-9S18Pp/tsserver.log --locale en --validateDefaultNpmLocation
Info 3    [10:27:53.675] Platform: darwin NodeVersion: 16 CaseSensitive: false
Info 4    [10:27:53.675] ServerMode: undefined syntaxOnly: false hasUnknownServerMode: undefined
Info 5    [10:27:53.688] Host information neovim
Info 6    [10:27:53.689] Format host information updated
Info 7    [10:27:53.691] Search path: /Users/myUser/myHugeProj/src/component
Info 8    [10:27:53.692] For info: /Users/myUser/myHugeProj/src/component.tsx :: Config file name: /Users/myUser/myHugeProj/tsconfig.json
Info 9    [10:27:53.693] Creating configuration project /Users/myUser/myHugeProj/tsconfig.json
Info 10   [10:27:53.708] Config: /Users/myUser/myHugeProj/tsconfig.json : {
 "rootNames": [],
 "options": {
  "rootDir": "/Users/myUser/myHugeProj",
  "sourceMap": true,
  "declaration": false,
  "moduleResolution": 2,
  "emitDecoratorMetadata": true,
  "experimentalDecorators": true,
  "importHelpers": true,
  "target": 2,
  "module": 99,
  "lib": [
   "lib.dom.d.ts",
   "lib.dom.iterable.d.ts",
   "lib.esnext.d.ts"
  ],
  "skipLibCheck": true,
  "skipDefaultLibCheck": true,
  "baseUrl": "/Users/myUser/myHugeProj",
  "paths": {
   "@myCompany/myCompanyFn": [
    "libs/org/myCompany/platform/fn/src/index.ts"
   ],
   "@myCompany/myCompanyRouter": [
    "libs/org/myCompany/platform/router/src/index.ts"
   ],
   "*": [
    "src/app/*"
   ]
  },
  "pathsBasePath": "/Users/myUser/myHugeProj",
  "jsx": 4,
  "allowJs": true,
  "esModuleInterop": true,
  "allowSyntheticDefaultImports": true,
  "forceConsistentCasingInFileNames": true,
  "strict": true,
  "noImplicitOverride": false,
  "noPropertyAccessFromIndexSignature": false,
  "noImplicitReturns": false,
  "noFallthroughCasesInSwitch": true,
  "resolveJsonModule": true,
  "types": [
   "node",
   "jest",
   "facebook-js-sdk",
   "segment-analytics"
  ],
  "configFilePath": "/Users/myUser/myHugeProj/tsconfig.json"
 },
 "projectReferences": [
  {
   "path": "/Users/myUser/myHugeProj/tsconfig.app.json",
   "originalPath": "./tsconfig.app.json"
  },
  {
   "path": "/Users/myUser/myHugeProj/tsconfig.spec.json",
   "originalPath": "./tsconfig.spec.json"
  },
  {
   "path": "/Users/myUser/myHugeProj/.storybook/tsconfig.json",
   "originalPath": "./.storybook/tsconfig.json"
  }
 ]
}
Info 11   [10:27:53.709] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.json
Info 12   [10:27:57.149] Config: /Users/myUser/myHugeProj/tsconfig.app.json : {
  // Same config as above.
}
Info 13   [10:27:59.886] Config: /Users/myUser/myHugeProj/tsconfig.spec.json : {
  // Same config as above.
}
}
Info 14   [10:28:01.770] Config: /Users/myUser/myHugeProj/.storybook/tsconfig.json : {
  // Probably not important here.
}
Info 15   [10:28:01.779] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.json Version: 1 structureChanged: true structureIsReused:: Not Elapsed: 8069.852646008134ms
Info 16   [10:28:01.779] Different program with same set of files
Info 17   [10:28:01.803] Creating configuration project /Users/myUser/myHugeProj/tsconfig.app.json
Info 18   [10:28:02.597] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 19   [10:28:22.733] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 1 structureChanged: true structureIsReused:: Not Elapsed: 20135.904230996966ms
Info 20   [10:28:22.734] Project '/Users/myUser/myHugeProj/tsconfig.app.json' (Configured)
Info 21   [10:28:22.734] 	Files (8548)

Info 22   [10:28:22.734] -----------------------------------------------
Info 23   [10:28:22.815] AutoImportProviderProject: attempted to add more than 10 dependencies. Aborting.

// Write: import { mybla } from "/src/constants";

Info 24   [10:30:06.201] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 25   [10:30:06.636] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 2 structureChanged: false structureIsReused:: Completely Elapsed: 435.0119200050831ms
Info 26   [10:30:06.636] Different program with same set of files
Info 27   [10:30:06.637] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 28   [10:30:06.958] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 3 structureChanged: false structureIsReused:: Completely Elapsed: 320.26086500287056ms
Info 29   [10:30:06.958] Different program with same set of files
Info 30   [10:30:06.980] getCompletionData: Get current token: 0.05790300667285919
·
·
·
Info 41   [10:30:07.584] Different program with same set of files
Info 42   [10:30:07.591] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 43   [10:30:07.863] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 6 structureChanged: false structureIsReused:: Completely Elapsed: 271.9172009974718ms
Info 44   [10:30:07.863] Different program with same set of files
Info 45   [10:30:07.866] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 46   [10:30:08.127] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 7 structureChanged: false structureIsReused:: Completely Elapsed: 261.01103200018406ms
Info 47   [10:30:08.127] Different program with same set of files

// The constants.ts file is saved.

Info 48   [10:37:59.356] DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/4913 :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
Info 49   [10:37:59.361] Elapsed:: 5.001938000321388ms DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/4913 :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
Info 50   [10:37:59.361] DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/constants.ts :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
Info 51   [10:37:59.363] Elapsed:: 0.576881006360054ms DirectoryWatcher:: Triggered with /users/myUser/myHugeProj/src/constants/constants.ts :: WatchInfo: /users/myUser/myHugeProj 1 undefined Config: /Users/myUser/myHugeProj/tsconfig.app.json WatchType: Wild card directory
·
·
·
Info 72   [10:38:01.201] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 73   [10:38:01.222] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 8 structureChanged: false structureIsReused:: Completely Elapsed: 20.91754199564457ms
Info 74   [10:38:01.222] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.json
Info 75   [10:38:02.110] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.json Version: 2 structureChanged: false structureIsReused:: Not Elapsed: 888.2716780006886ms
Info 76   [10:38:02.110] Before ensureProjectForOpenFiles:
Info 77   [10:38:02.110] After ensureProjectForOpenFiles:
Info 78   [10:38:02.111] got projects updated in background, updating diagnostics for /Users/myUser/myHugeProj/src/component.tsx

// THIS IS WHERE THE WAITING HAPPENS!!! Between 3 to 6 minutes.

Info 79   [10:42:52.259] FileWatcher:: Triggered with /Users/myUser/myHugeProj/src/constants.ts 1:: WatchInfo: /Users/myUser/myHugeProj/src/constants.ts 500 undefined WatchType: Closed Script info
Info 80   [10:42:52.259] Elapsed:: 0.483830988407135ms FileWatcher:: Triggered with /Users/myUser/myHugeProj/src/constants.ts 1:: WatchInfo: /Users/myUser/myHugeProj/src/constants.ts 500 undefined WatchType: Closed Script info
Info 81   [10:42:52.510] Starting updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json
Info 82   [10:42:52.865] Finishing updateGraphWorker: Project: /Users/myUser/myHugeProj/tsconfig.app.json Version: 9 structureChanged: false structureIsReused:: Completely Elapsed: 354.6076129972935ms
Info 83   [10:42:52.865] Different program with same set of files
Info 84   [10:42:54.760] Before ensureProjectForOpenFiles:
Info 85   [10:42:54.760] After ensureProjectForOpenFiles:
Info 86   [10:42:54.760] got projects updated in background, updating diagnostics for /Users/myUser/myHugeProj/src/component.tsx
```

**Comment by rchl** [comment]
VSCode runs two instances of tsserver so it's able to provide some functionality while the project is loading.

I'm experimenting with porting that code and might have something ready soon.

**Comment by yusuphgammut** [comment]
@rchl I've done a few tests, and seems that the problem is solved now. I'll working with this new release to see if everything is working as expected.

Thanks a lot for your time and effort in solving this!
